tests/test_articlehistory.py

```diff
@@ -29,5 +30,76 @@ def test___init__():
     with pytest.raises(KeyError):
         raise KeyError("Data not found")
 
+
+def test_filter_by_tags():
+    '''tests filter by tags function'''
+    art = ArticleHistory("Techno", None, None, ["mobile edit"])
```

Comment:
Leaving timestamps unspecified means our tests are not consistent - this will pull down a batch of the most recent revisions made to the techno article, and there may or may not be revisions tagged with "mobile edit" in this batch depending on when the test is run. Instead, consider adding timestamp(s) to the test case to make sure the program's behavior is consistent no matter when the test is run.

Reply:
(this applies to all cases in this test function)